src/app/clusters/scenes-server/SceneTableImpl.cpp

```diff
@@ -52,26 +50,28 @@ using SceneData       = DefaultSceneTableImpl::SceneData;
 // byte value, 1 byte end struct. 8 Bytes leaves space for potential increase in count_value size.
 static constexpr size_t kPersistentBufferSceneCountBytes = 8;
 
-struct GlobalSceneCount : public PersistentData<kPersistentBufferSceneCountBytes>
+struct EndpointSceneCount : public PersistentData<kPersistentBufferSceneCountBytes>
 {
-    uint8_t count_value = 0;
+    EndpointId endpoint_id = kInvalidEndpointId;
+    uint8_t count_value    = 0;
 
-    GlobalSceneCount(uint8_t count = 0) : count_value(count) {}
-    ~GlobalSceneCount() {}
+    EndpointSceneCount(EndpointId endpoint, uint8_t count = 0) : endpoint_id(endpoint), count_value(count) {}
+    ~EndpointSceneCount() {}
 
     void Clear() override { count_value = 0; }
 
     CHIP_ERROR UpdateKey(StorageKeyName & key) override
     {
-        key = DefaultStorageKeyAllocator::GlobalSceneCountKey();
+        VerifyOrReturnError(kInvalidEndpointId != endpoint_id, CHIP_ERROR_INVALID_ARGUMENT);
+        key = DefaultStorageKeyAllocator::EndpointSceneCountKey(endpoint_id);
```

Comment:
How does this interact with dynamic endpoints going away?  What cleans up the storage?